File: code_blocks/integrations/third-party-integrations/onesignal_14.dart
import 'package:onesignal_flutter/onesignal_flutter.dart';
import 'package:purchases_flutter/purchases_flutter.dart';

Future<void> initOneSignalRevenueCat() async {
  await Purchases.configure(PurchasesConfiguration('<revenuecat_api_key>'));

  OneSignal.initialize('<onesignal_app_id>');
  OneSignal.User.addObserver((state) {
    final onesignalId = state.current.onesignalId;
    if (onesignalId != null) {
      Purchases.setOnesignalUserID(onesignalId);
    }
  });

  final onesignalId = OneSignal.User.onesignalId;
  if (onesignalId != null) {
    await Purchases.setOnesignalUserID(onesignalId);
  }
}

File: code_blocks/integrations/third-party-integrations/onesignal_14.ts
import OneSignal, { UserChangedState } from 'react-native-onesignal';
import Purchases from '@revenuecat/purchases-react-native';

export async function initOneSignalRevenueCat() {
  await Purchases.configure({ apiKey: '<revenuecat_api_key>' });

  OneSignal.initialize('<onesignal_app_id>');
  OneSignal.User.addObserver((state: UserChangedState) => {
    const onesignalId = state.current.onesignalId;
    if (onesignalId) {
      Purchases.setOnesignalUserID(onesignalId);
    }
  });

  const onesignalId = OneSignal.User.onesignalId;
  if (onesignalId) {
    await Purchases.setOnesignalUserID(onesignalId);
  }
}
